plugins/modules/nsxt_logical_routers.py

```diff
@@ -274,21 +273,105 @@ def update_params_with_id (module, manager_url, mgr_username, mgr_password, vali
 
+
+def get_id_from_display_name_uplink(module, manager_url, mgr_username, mgr_password, validate_certs, endpoint,
+                                    uplink_profiles_display_names):
+    uplink_profile_display_name1 = uplink_profiles_display_names[0]
+    uplink_profile_display_name2 = uplink_profiles_display_names[1]
+    try:
+        (rc, resp) = request(manager_url + endpoint, headers=dict(Accept='application/json'),
+                             url_username=mgr_username, url_password=mgr_password, validate_certs=validate_certs,
+                             ignore_errors=True)
+    except Exception as err:
+        module.fail_json(msg='Error accessing id for uplink display names %s, %s. Error [%s]' % (
+        uplink_profile_display_name1, uplink_profile_display_name2, to_native(err)))
+
+    uplink_profile_ids = []
+
+    for result in resp['results']:
+        if result.__contains__('display_name') and (result['display_name'] == uplink_profile_display_name1 or result[
+            'display_name'] == uplink_profile_display_name2):
+            uplink_profile_ids.append(result['id'])
+    if uplink_profile_ids is None or len(uplink_profile_ids) < 2:
+        module.fail_json(msg='No id exists with uplink display name %s, %s' % (
+        uplink_profile_display_name1, uplink_profile_display_name2))
+    return uplink_profile_ids
+
+
 def check_for_update(module, manager_url, mgr_username, mgr_password, validate_certs, logical_router_with_ids):
     existing_logical_router = get_lr_from_display_name(module, manager_url, mgr_username, mgr_password, validate_certs, logical_router_with_ids['display_name'])
     if existing_logical_router is None:
         return False
-    if existing_logical_router.__contains__('edge_cluster_id') and logical_router_with_ids.__contains__('edge_cluster_id') and \
-        existing_logical_router['edge_cluster_id'] != logical_router_with_ids['edge_cluster_id']:
+    if existing_logical_router.__contains__('tags') and not logical_router_with_ids.__contains__('tags'):
+        return True
+    if not existing_logical_router.__contains__('tags') and logical_router_with_ids.__contains__('tags'):
+        return True
+    if existing_logical_router.__contains__('tags') and logical_router_with_ids.__contains__('tags') and (
+            not compareTags(existing_logical_router, logical_router_with_ids)):
         return True
-    if existing_logical_router.__contains__('advanced_config') and logical_router_with_ids.__contains__('advanced_config'):
-        if existing_logical_router['advanced_config'].__contains__('internal_transit_networks') and logical_router_with_ids['advanced_config'].__contains__('internal_transit_networks') and \
-            existing_logical_router['advanced_config']['internal_transit_networks'] != logical_router_with_ids['advanced_config']['internal_transit_networks']:
+    if existing_logical_router.__contains__('edge_cluster_id') and logical_router_with_ids.__contains__(
```

Comment:
Bug ID Fixed - 2654045 - Updation of certain fields are not supported by 'nsxt_logical_routers' module